samples/src/main/java/com/example/bigquery/AddColumnLoadAppend.java

```diff
@@ -0,0 +1,102 @@
+import com.google.cloud.bigquery.LoadJobConfiguration;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableId;
+import com.google.common.collect.ImmutableList;
+import java.util.UUID;
+
+public class AddColumnLoadAppend {
+
+  public static void runAddColumnLoadAppend() {
+    // TODO(developer): Replace these variables before running the sample.
+    String datasetName = "MY_DATASET_NAME";
+    String tableName = "MY_TABLE_NAME";
+    String sourceUri = "/path/to/file.csv";
+    addColumnLoadAppend(datasetName, tableName, sourceUri);
+  }
+
+  public static void addColumnLoadAppend(String datasetName, String tableName, String sourceUri) {
+    try {
+      // Initialize client that will be used to send requests. This client only needs to be created
+      // once, and can be reused for multiple requests.
+      BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+
+      TableId tableId = TableId.of(datasetName, tableName);
+      Table table = bigquery.getTable(tableId);
+
+      // Add a new column to a BigQuery table while appending rows via a load job.
+      // 'REQUIRED' fields cannot  be added to an existing schema, so the additional column must be
+      // 'NULLABLE'.
+      Schema newSchema =
+          Schema.of(
+              Field.newBuilder("name", LegacySQLTypeName.STRING)
+                  .setMode(Field.Mode.REQUIRED)
+                  .build(),
+              // Adding below additional column during the load job
+              Field.newBuilder("post_abbr", LegacySQLTypeName.STRING)
+                  .setMode(Field.Mode.NULLABLE)
+                  .build());
+
+      LoadJobConfiguration loadJobConfig =
+          LoadJobConfiguration.builder(tableId, sourceUri)
+              .setFormatOptions(FormatOptions.csv())
+              .setWriteDisposition(WriteDisposition.WRITE_APPEND)
+              .setSchema(newSchema)
+              .setSchemaUpdateOptions(ImmutableList.of(SchemaUpdateOption.ALLOW_FIELD_ADDITION))
+              .build();
+
+      // Create a job ID so that we can safely retry.
+      JobId jobId = JobId.of(UUID.randomUUID().toString());
+      Job loadJob = bigquery.create(JobInfo.newBuilder(loadJobConfig).setJobId(jobId).build());
+      System.out.println(loadJob.getJobId());
+
+      // Load data from a GCS parquet file into the table
+      // Blocks until this load table job completes its execution, either failing or succeeding.
+      Job completedJob = loadJob.waitFor();
+
+      // Check for errors
+      if (completedJob == null) {
+        throw new RuntimeException("Job no longer exists");
+      } else if (completedJob.getStatus().getError() != null) {
+        // You can also look at queryJob.getStatus().getExecutionErrors() for all
+        // errors, not just the latest one.
+        throw new RuntimeException(loadJob.getStatus().getError().toString());
```

Comment:
nit: It there error throwable? e.g. `throw loadJob.getStatus().getError().toString()`? Else use a checked error.

Reply:
yes - in fact, it helped me debug when I wasn't able to figure out what was wrong with the job.

Reply:
What I meant was can you throw the Error directly - using `throw loadJob.getStatus().getError()`?

Otherwise you really shouldn't throw a unchecked exceptions like this - checked exceptions are preferred. (A regular `Exception` is considered checked, while `RuntimeExceptions` are unchecked.)

Reply:
Oh I see... I'm updating this to be print it out since `loadJob.getStatus().getError()` is a String instead of Throwable.

Reply:
We want to throw - it should just be a checked exception.